homeassistant/components/apple_tv/config_flow.py

```diff
@@ -0,0 +1,411 @@
+        if user_input is not None:
+            self.target_device = user_input[DEVICE_INPUT]
+            try:
+                await self.async_find_device()
+            except DeviceNotFound:
+                errors["base"] = "device_not_found"
+            except DeviceAlreadyConfigured:
+                errors["base"] = "device_already_configured"
+            except exceptions.NoServiceError:
+                errors["base"] = "no_usable_service"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(
+                    self.atv.identifier, raise_on_progress=False
+                )
+                return await self.async_step_confirm()
+
+            # Use whatever the user entered as default value
+            default_suggestion = self.unique_id
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(DEVICE_INPUT, default=default_suggestion): str}
+            ),
+            errors=errors,
+            description_placeholders={"devices": self._devices_str()},
+        )
+
+    async def async_step_zeroconf(self, discovery_info):
+        """Handle device found via zeroconf."""
+        service_type = discovery_info[CONF_TYPE]
+        properties = discovery_info["properties"]
+
+        if service_type == "_mediaremotetv._tcp.local.":
+            identifier = properties["UniqueIdentifier"]
+            name = properties["Name"]
+        elif service_type == "_touch-able._tcp.local.":
+            identifier = discovery_info["name"].split(".")[0]
+            name = properties["CtlN"]
+        else:
+            return self.async_abort(reason="unrecoverable_error")
+
+        await self.async_set_unique_id(identifier)
+        self._abort_if_unique_id_configured()
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["identifier"] = self.unique_id
+        self.context["title_placeholders"] = {"name": name}
+        self.target_device = identifier
+        return await self.async_find_device_wrapper(self.async_step_confirm)
+
+    async def async_find_device_wrapper(self, next_func, allow_exist=False):
+        """Find a specific device and call another function when done.
+
+        This function will do error handling and bail out when an error
+        occurs.
+        """
+        try:
+            await self.async_find_device(allow_exist)
+        except DeviceNotFound:
+            return self.async_abort(reason="device_not_found")
+        except DeviceAlreadyConfigured:
+            return self.async_abort(reason="already_configured_device")
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            return self.async_abort(reason="unrecoverable_error")
```

Comment:
Please use unknown error key and a reference here.

Reply:
Only here or replace `unrecoverable_error` in a places?

Reply:
In all places where we catch a broad exception and return error or abort.

Reply:
Alright, should be fixed now.